encoding/ccf/decode_type.go

```diff
@@ -436,6 +440,46 @@ func (d *Decoder) decodeDictType(
 	return cadence.NewMeteredDictionaryType(d.gauge, keyType, elementType), nil
 }
 
+// decodeInclusiveRangeType decodes inclusiverange-type or inclusiverange-type-value as
+// language=CDDL
+// inclusiverange-type =
+//
+//	; cbor-tag-inclusiverange-type
+//	#6.145([
+//	    element-type: inline-type
+//	])
+//
+// inclusiverange-type-value =
+//
+//	; cbor-tag-inclusiverange-type-value
+//	#6.194([
+//	    element-type: type-value
+//	])
+//
+// NOTE: decodeTypeFn is responsible for decoding inline-type or type-value.
+func (d *Decoder) decodeInclusiveRangeType(
+	types *cadenceTypeByCCFTypeID,
+	decodeTypeFn decodeTypeFn,
+) (cadence.Type, error) {
+	// Decode array head of length 1.
+	err := decodeCBORArrayWithKnownSize(d.dec, 1)
+	if err != nil {
+		return nil, err
+	}
+
+	// element 0: element type (inline-type or type-value)
+	elementType, err := decodeTypeFn(types)
+	if err != nil {
+		return nil, err
+	}
+
+	if elementType == nil {
+		return nil, errors.New("unexpected nil type as inclusiverange element type")
+	}
```

Comment:
This seems like unnecessary. If I pass a nil elementType, it leads to an error in the above `decodeTypeFn` and we return in the `err != nil` check. It might still be OK to keep this check here as we have similar code in other decode methods such as for dictionary.